urldae1c57e59_mustafaakin_izuna_mapeditor_tgz-pJ8-in_mustafaak_izuna_actions_ActionTestJ8/src/in/mustafaak/izuna/waves/set/B.java

```diff
@@ -0,0 +1,104 @@
+package in.mustafaak.izuna.waves.set;
+
+
+import in.mustafaak.izuna.actions.*;
+import in.mustafaak.izuna.meta.WaveInfo;
+import in.mustafaak.izuna.waves.CircularInvade;
+import in.mustafaak.izuna.waves.Invade;
+import in.mustafaak.izuna.waves.RowColEnter;
+
+public class B {
+	public static WaveInfo w1() {
+		WaveInfo w = new RowColEnter(RowColEnter.FROM_UP, 100, 200, 1, 4, 150, 0, "b1", 2000);
+		Action.wiggle(w, 100, 0, 2000);
+		return w;
+	}
+
+	public static WaveInfo w2() {
+		WaveInfo w = new RowColEnter(RowColEnter.FROM_LEFT, 100, 50, 1, 4, 150, 0, "b1", 2000);
+		Action.wiggle(w, 150, 0, 2000);
+		return w;
+	}
+
+	public static WaveInfo w3() {
+		WaveInfo w = new RowColEnter(RowColEnter.FROM_RIGHT, 100, 50, 1, 4, 150, 0, "b1", 2000);
+		Action.wiggle(w, -150, 0, 2000);
+		return w;
+	}
+
+	public static WaveInfo w4() {
+		WaveInfo w = new Invade(4, -200, -200, 500, 700, 100, 600, 400, 200, "a2", 4000);
+		return w;
+	}
+
+	public static WaveInfo w5() {
+		WaveInfo w = new WaveInfo();
+
+		Start s = new Start(-200, 100);
+		Loop l = new Loop();
+		CrossAndUp c = new CrossAndUp(200, 300, 3000);
+		for (int i = 0; i < 4; i++) {
+			w.putEnemy("a2", s, l, new Delay(i * 1500), c, c, c, c, c);
+
+		}
+		return w;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 35:44, 48:57.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.